src/FlagProvider.tsx

```diff
@@ -40,7 +42,10 @@ const FlagProvider: React.FC<IFlagProvider> = ({
   });
 
   React.useEffect(() => {
-    client.current.start();
+    const shouldStartClient = startClient || !unleashClient;
```

Comment:
So basically if there is no valid unleash client, this will start client, no matter what startClient is set?